lib/db/schema/patch-022-023.sql

```diff
@@ -0,0 +1,263 @@
+  IN uaOS VARCHAR(255),
+  IN uaOSVersion VARCHAR(255),
+  IN uaDeviceType VARCHAR(255),
+  IN tokenVerified BOOLEAN,
+  IN tokenVerificationId BINARY(32)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  INSERT INTO sessionTokens(
+    tokenId,
+    tokenData,
+    uid,
+    createdAt,
+    uaBrowser,
+    uaBrowserVersion,
+    uaOS,
+    uaOSVersion,
+    uaDeviceType,
+    lastAccessTime
+  )
+  VALUES(
+    tokenId,
+    tokenData,
+    uid,
+    createdAt,
+    uaBrowser,
+    uaBrowserVersion,
+    uaOS,
+    uaOSVersion,
+    uaDeviceType,
+    createdAt
+  );
+
+  IF tokenVerified = FALSE THEN
+    INSERT INTO unverifiedTokens(
+      tokenVerificationId,
+      tokenId
+    )
+    VALUES(
+      tokenVerificationId,
+      tokenId
+    );
+  END IF;
+
+  COMMIT;
+END;
+
+-- Update createKeyFetchToken stored procedure to
+-- insert into unverifiedTokens.
+CREATE PROCEDURE `createKeyFetchToken_2` (
+  IN tokenId BINARY(32),
+  IN authKey BINARY(32),
+  IN uid BINARY(16),
+  IN keyBundle BINARY(96),
+  IN createdAt BIGINT UNSIGNED,
+  IN tokenVerified BOOLEAN,
+  IN tokenVerificationId BINARY(32)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  INSERT INTO keyFetchTokens(
+    tokenId,
+    authKey,
+    uid,
+    keyBundle,
+    createdAt
+  )
+  VALUES(
+    tokenId,
+    authKey,
+    uid,
+    keyBundle,
+    createdAt
+  );
+
+  IF tokenVerified = FALSE THEN
+    INSERT INTO unverifiedTokens(
+      tokenVerificationId,
+      tokenId
+    )
+    VALUES(
+      tokenVerificationId,
+      tokenId
+    );
+  END IF;
+
+  COMMIT;
+END;
+
+-- Update deleteSessionToken stored procedure to
+-- conditionally delete from unverifiedTokens.
+CREATE PROCEDURE `deleteSessionToken_2` (
+  IN tokenIdArg BINARY(32)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  DELETE FROM sessionTokens WHERE tokenId = tokenIdArg;
+  DELETE FROM unverifiedTokens WHERE tokenId = tokenIdArg;
+
+  COMMIT;
+END;
+
+-- Update deleteKeyFetchToken stored procedure to
+-- conditionally delete from unverifiedTokens.
+CREATE PROCEDURE `deleteKeyFetchToken_2` (
+  IN tokenIdArg BINARY(32)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  DELETE FROM keyFetchTokens WHERE tokenId = tokenIdArg;
+  DELETE FROM unverifiedTokens WHERE tokenId = tokenIdArg;
+
+  COMMIT;
+END;
+
+-- Add stored procedure for fetching sessionToken
+-- with its verified state.
+CREATE PROCEDURE `sessionTokenVerified_1` (
+  IN tokenIdArg BINARY(32)
+)
+BEGIN
+  DECLARE tokenVerified BOOLEAN;
+
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  SET tokenVerified = (
+    CASE WHEN EXISTS(
+      SELECT * FROM unverifiedTokens
```

Comment:
nit: if we `SELECT 1` or `SELECT tokenId` here I think this will be an index only query and faster